From: Georges Khaznadar <georgesk@debian.org>
Date: Sat, 3 Sep 2016 17:57:10 +0200
Subject: Modifying the Exec cmd in the desktop-files

Fixup the *.desktop files and add a file handler to the Exec key.
https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html
---
 resources/linux/mime/applications/bitmap2component.desktop |  4 +++-
 resources/linux/mime/applications/cvpcb.desktop            | 10 ++++++++++
 resources/linux/mime/applications/eeschema.desktop         |  1 +
 resources/linux/mime/applications/gerbview.desktop         |  5 +++--
 resources/linux/mime/applications/kicad.desktop            |  2 ++
 resources/linux/mime/applications/pcbcalculator.desktop    |  5 +++--
 resources/linux/mime/applications/pcbnew.desktop           |  1 +
 7 files changed, 23 insertions(+), 5 deletions(-)
 create mode 100644 resources/linux/mime/applications/cvpcb.desktop

diff --git a/resources/linux/mime/applications/bitmap2component.desktop b/resources/linux/mime/applications/bitmap2component.desktop
index 07da619..aa2b78c 100644
--- a/resources/linux/mime/applications/bitmap2component.desktop
+++ b/resources/linux/mime/applications/bitmap2component.desktop
@@ -4,7 +4,9 @@ Terminal=false
 Icon=bitmap2component
 Type=Application
 Categories=Development;Electronics;
-Exec=bitmap2component
+Exec=bitmap2component %f
+Keywords=Bitmap;Converter;
+MimeType=application/x-bitmap2component-project;
 Name=Bitmap to Component Converter
 GenericName=Bitmap to Component Converter
 GenericName[it]=Convertitore da bitmap a componente
diff --git a/resources/linux/mime/applications/cvpcb.desktop b/resources/linux/mime/applications/cvpcb.desktop
new file mode 100644
index 0000000..eadbc64
--- /dev/null
+++ b/resources/linux/mime/applications/cvpcb.desktop
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Categories=Development;Electronics;
+Comment=Assign footprints to symbols (part of KiCad)
+Exec=cvpcb %f
+GenericName=EDA Suite
+Icon=cvpcb
+MimeType=application/x-cvpcb-project;
+Name=CvPcb
+Type=Application
+Name[en_US]=CvPcb
diff --git a/resources/linux/mime/applications/eeschema.desktop b/resources/linux/mime/applications/eeschema.desktop
index 9f6cd18..845ecea 100644
--- a/resources/linux/mime/applications/eeschema.desktop
+++ b/resources/linux/mime/applications/eeschema.desktop
@@ -7,6 +7,7 @@ Categories=Development;Electronics;
 Exec=eeschema %f
 MimeType=application/x-kicad-schematic;
 Name=Eeschema (Standalone)
+Keywords=Schematic;Editor;
 GenericName=Electronic schematic capture
 GenericName[fr]=Saisie de schéma électronique
 GenericName[it]=Disegno di schemi elettrici
diff --git a/resources/linux/mime/applications/gerbview.desktop b/resources/linux/mime/applications/gerbview.desktop
index 268b46e..bf10044 100644
--- a/resources/linux/mime/applications/gerbview.desktop
+++ b/resources/linux/mime/applications/gerbview.desktop
@@ -4,10 +4,11 @@ Terminal=false
 Icon=gerbview
 Type=Application
 Categories=Development;Electronics;
-Exec=gerbview
+Exec=gerbview %f
 MimeType=application/x-gerber;application/x-excellon
+Keywords=PCB;Viewer;
 Name=GerbView
 GenericName=Gerber File Viewer
 GenericName[it]=Visualizzatore file Gerber
 Comment=View Gerber files
-Comment[it]=Visualizza file in formato Gerber
\ No newline at end of file
+Comment[it]=Visualizza file in formato Gerber
diff --git a/resources/linux/mime/applications/kicad.desktop b/resources/linux/mime/applications/kicad.desktop
index 5cc69ba..03291c4 100644
--- a/resources/linux/mime/applications/kicad.desktop
+++ b/resources/linux/mime/applications/kicad.desktop
@@ -5,9 +5,11 @@ Icon=kicad
 Type=Application
 Categories=Development;Electronics;
 Exec=kicad %f
+Keywords=Kicad;Projectmanager;
 MimeType=application/x-kicad-project;
 Name=KiCad
 GenericName=EDA Suite
+GenericName[fr]=Suite logicielle de conception électronique
 GenericName[it]=CAD elettronico
 Comment=Electronic Design Automation suite
 Comment[fr]=Suite logicielle de conception électronique
diff --git a/resources/linux/mime/applications/pcbcalculator.desktop b/resources/linux/mime/applications/pcbcalculator.desktop
index 7158e04..690787b 100644
--- a/resources/linux/mime/applications/pcbcalculator.desktop
+++ b/resources/linux/mime/applications/pcbcalculator.desktop
@@ -1,10 +1,11 @@
 [Desktop Entry]
 Version=1.0
 Terminal=false
+Categories=Development;Electronics;
+Exec=pcb_calculator %f
+Keywords=PCB;Caculator;
 Icon=pcbcalculator
 Type=Application
-Categories=Development;Electronics;
-Exec=pcb_calculator
 Name=PCB Calculator
 GenericName=PCB Calculator
 Comment=Calculator for various electronics-related computations
diff --git a/resources/linux/mime/applications/pcbnew.desktop b/resources/linux/mime/applications/pcbnew.desktop
index 57e5c50..93eb91c 100644
--- a/resources/linux/mime/applications/pcbnew.desktop
+++ b/resources/linux/mime/applications/pcbnew.desktop
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Version=1.0
 Terminal=false
+Keywords=PCB;Circuit;Layout;Design;Editor;Footprint;
 Icon=pcbnew
 Type=Application
 Categories=Development;Electronics;
